src/main/java/com/customer/service/CustomerServiceImpl.java

```diff
@@ -50,19 +60,56 @@ public Customer createCustomer(Customer customer) {
 	}
 
 	@Override
-	public CustomerResponse findCustomerById(int id) {
-		// TODO Auto-generated method stub
+	public CustomerResponse findCustomerById(int id, Principal principal) {
+		String userName = principal.getName(); // get logged in username
 		Optional<Customer> customer = customerRepository.findById(id);
+
 		if (customer.isPresent()) {
-			CustomerResponse customerResponse = CustomerResponse.builder().id(customer.get().getId())
-					.firstName(customer.get().getFirstName()).lastName(customer.get().getLastName())
-					.joiningDate(customer.get().getJoiningDate()).expiryDate(customer.get().getExpiryDate())
-					.address(customer.get().getAddress()).build();
-			customerResponse.setStatus(customer.get().getJoiningDate(), customer.get().getExpiryDate());
-			return customerResponse;
-		}
-		else
+			if (customer.get().getUserName().equals(userName)) {
+				CustomerResponse customerResponse = CustomerResponse.builder().id(customer.get().getId())
+						.firstName(customer.get().getFirstName()).lastName(customer.get().getLastName())
+						.joiningDate(customer.get().getJoiningDate()).expiryDate(customer.get().getExpiryDate())
+						.address(customer.get().getAddress()).build();
+				customerResponse.setStatus(customer.get().getJoiningDate(), customer.get().getExpiryDate());
+				return customerResponse;
+			} else
+				throw new ForbiddenException("Invalid Credentials");
+		} else
 			throw new CustomerNotFoundException("Customer not found for id = ", id);
 	}
-}
 
+	@Override
+	public Customer updateCustomerDetails(int id, Customer customer, Principal principal) {
+		String userName = principal.getName(); // get logged in username
+		Optional<Customer> customerRes = customerRepository.findById(id);
+		if (customerRes.isPresent()) {
+			if (customerRes.get().getUserName().equals(userName) && customerRes.get().getId() == id) {
```

Comment:
there is no need to check the 2nd condition here right ?

`customerRes.get().getId() == id` <-- thats exactly what you are fetching .. why check again

Reply:
Done.